### PR TITLE
feat: stores autosuggest added

### LINKED ISSUE #3301: Stores autosuggest
The Stores taxonomy is available: https://static.openfoodfacts.org/data/taxonomies/stores.json
It can be loaded for stores autocomplete in add/edit mode.

### PR DESCRIPTION
**Pull request by kartikaysharma01**
####  Description

Added stores taxonomy.

#### Related issues

Fixes #3301

#### Screenshots

stores autosuggest

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/entities/store/StoreResponse.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class StoreResponse (
+    private var code: String,
+    private var names: Map<String, String>,
+    private val wikiDataCode: String? = null
+) : EntityResponse<Store> {
+    override fun map(): Store {
+        val store: Store
+        if (wikiDataCode != null) {
+            store = Store(code, arrayListOf(), wikiDataCode)
+            names.forEach { (key, value) ->
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/entities/store/StoreResponse.kt:13]: Can we use .apply{} here? And then inline the return.

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/entities/store/StoreResponse.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+                store.names.add(StoreName(store.tag, key, value, wikiDataCode))
+            }
+        } else {
+            store = Store(code, arrayListOf())
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/entities/store/StoreResponse.kt]: Can we use .apply{} here? And then inline the return

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/repositories/ProductRepository.kt [hunk 3]
```diff
@@ -423,6 +438,31 @@ object ProductRepository {
...
 
+    /**
+     * Stores saving to local database
+     *
+     * @param stores The list of stores to be saved.
+     *
+     *
+     * store and storeName has One-To-Many relationship, therefore we need to save them separately.
+     */
+    private fun saveStore(stores: List<Store>) {
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/repositories/ProductRepository.kt]: Better rename it to "saveStores"

**Comment by VaiTon** [comment]
Good work! Just some little things

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/entities/store/StoreResponse.kt (hunk 1 above)
Comment (kartikaysharma01) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/models/entities/store/StoreResponse.kt:13]: We can use .also{}

**Comment by kartikaysharma01** [comment]
@VaiTon I have pushed the requested changes. Thanks.

**Review by VaiTon — APPROVED** [review]

**Comment by VaiTon** [comment]
Thank you 🎉

**Pull request merged**